cli/src/main/scala-2.12/coursier/cli/Helper.scala

```diff
@@ -273,11 +272,37 @@ class Helper(
       intransitiveModVerCfgErrors.map("  "+_).mkString("\n")
   }
 
-  val transitiveDeps: Seq[Dependency] =
+  val transitiveDepsWithExtraParams: Seq[(Dependency, Map[String, String])] =
   // FIXME Order of the dependencies is not respected here (scaladex ones go first)
-    scaladexDeps ++ normalDeps
+    scaladexDepsWithExtraParams ++ normalDepsWithExtraParams
+
+  val transitiveDeps: Seq[Dependency] = transitiveDepsWithExtraParams.map(dep => dep._1)
+
+  val allDependenciesWithExtraParams: Seq[(Dependency, Map[String, String])] =
+    transitiveDepsWithExtraParams ++ intransitiveDepsWithExtraParams
+
+  val allDependencies: Seq[Dependency] = allDependenciesWithExtraParams.map(dep => dep._1)
+
+  // Any dependencies with URIs should not be resolved with a pom so this is a
+  // hack to add all the deps with URIs to the FallbackDependenciesRepository
+  // which will be used during the resolve
+  val depsWithUrls: Map[(Module, String), (URL, Boolean)] = {
+    allDependenciesWithExtraParams
+      .filter(depWithExtraParams => depWithExtraParams._2.isDefinedAt("url"))
+      .map(depWithExtraParams =>
+        (depWithExtraParams._1.moduleVersion, (new URL(depWithExtraParams._2.getOrElse("url", "")), true))).toMap
```

Comment:
It's possible to check the presence of and get the `"url"` value in one pass, like
```scala
allDependenciesWithExtraParams
  .flatMap {
    case (dep, extraParams) =>
      extraParams.get("url").map { url =>
        dep.moduleVersion -> (new URL(url), true)
      }
  }
```